apps/core/declarativewebcontainer.cpp

```diff
@@ -496,17 +496,30 @@ void DeclarativeWebContainer::closeTab(int tabId)
 }
 
 int DeclarativeWebContainer::activateTab(int tabId, const QString &url)
+{
+    return requestTabWithOwner(tabId, url, 0);
+}
+
+int DeclarativeWebContainer::requestTabWithOwner(int tabId, const QString &url, uint ownerPid)
 {
     bool activated = m_model->activateTabById(tabId);
     if (!activated) {
         tabId = m_model->newTab(url);
+        // TODO: remove ownership of the tab if user changes URL
```

Comment:
Maybe navigation on the requested tab is fine and changing url should be nice as we really cannot assume much about usage of the requested tab. That said, if there's onOpenUrlRequested or user herself enters a url (function loadPage(url) of the Overlay) then I think it'd make sense to remove ownership.

How does that sound to you?

Reply:
Pushed a commit that might work for this purpose.